src/App.js

```diff
@@ -1,5 +1,44 @@
+import { createContext, useState, useEffect } from "react";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import { collection, getDocs } from "firebase/firestore";
+
+import { db } from "./firebase-config";
+import Navbar from "./components/Navbar";
+import Login from "./components/Login";
+import Users from "./components/Users";
+import Expense from "./components/Expense";
+import CreateExpense from "./components/Expense/CreateExpense";
+
+export const userContext = createContext();
+
 const App = () => {
-  return <div>App</div>;
+  const [users, setUsers] = useState([]);
+  const usersCollectionRef = collection(db, "users");
+
+  useEffect(() => {
+    const getUserData = async () => {
```

Comment:
This method should be in separate file and use it after importing from the file. This way we can use in multiple places if we need to.
Do it if possible as I am not sure about it

Reply:
Yes its possible. At broad level its become more complex. Become difficult to find where data came from. 
A suggestion from my side, we can divide these big component into small components.